### PR TITLE
Add playground for development

### PR DESCRIPTION
**Pull request by ghostdevv**
In the Svelte repo we have a playground for locally testing changes, but we don't here - this adds that. WDYT?

My main issue currently is that there are no types for most modules `@sveltejs/kit`, `$app/stores`, etc... I believe this is due to the `generate-dts` script not being run till kit is released - I don't know the best way to fix this. It's also an issue in the other packages so it's not an issue introduced by this pr from what I can tell.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Comment by benmccann** [comment]
I've personally never felt the need as I simply use the kit.svelte.dev site that's already checked in for testing, but I have no objection if others find it useful

**Comment by ghostdevv** [comment]
> I've personally never felt the need as I simply use the kit.svelte.dev site that's already checked in for testing, but I have no objection if others find it useful

I've also sometimes used that, but since there is a lot going on there I find it can be harder to test stuff sometimes

**Comment by dummdidumm** [comment]
I'm in favor of this change. I use kit.svelte.dev often for this but if also often gets in the way with its prerendering pipeline so I delete stuff temporarily there. This would avoid that.

**Comment by benmccann** [comment]
this PR should probably update https://github.com/sveltejs/kit/blob/master/CONTRIBUTING.md. but otherwise lgtm

**Comment by ghostdevv** [comment]
I added a note in the contributing.md @benmccann

RE: the types problem, is that just me? if not is there something we can do about it, or is it a ["this is fine"](https://tenor.com/en-GB/view/this-is-fine-gif-24177057) situation?

**Comment by gtm-nayan** [comment]
Maybe a good idea to also add a file with these contents:

```json
{
	"pnpm": {
		"overrides": {
			"@sveltejs/kit": "../kit/packages/kit",
			"@sveltejs/adapter-auto": "../kit/packages/adapter-auto",
			"@sveltejs/adapter-cloudflare": "../kit/packages/adapter-cloudflare",
			"@sveltejs/adapter-cloudflare-workers": "../kit/packages/adapter-cloudflare-workers",
			"@sveltejs/adapter-netlify": "../kit/packages/adapter-netlify",
			"@sveltejs/adapter-node": "../kit/packages/adapter-node",
			"@sveltejs/adapter-static": "../kit/packages/adapter-static",
			"@sveltejs/adapter-vercel": "../kit/packages/adapter-vercel"
		}
	}
}
```

so that there's an easily accessible copypasta when cloning repros. Not sure if there's a better place for it outside the playground folder

**Comment by ghostdevv** [comment]
> Maybe a good idea to also add a file with these contents:

We already have it in the CONTRIBUTING.md so maybe there? Though not sure if we would want to have a list of all packages imo

**Comment by gtm-nayan** [comment]
If I understood the description correctly, Geoff's PR to generate the types on postinstall should've fixed it.

**Comment by ghostdevv** [comment]
Yep #10301 fixed that problem! Though I do wonder if we should watch for changes - at any rate the playground itself is good to go

**Comment by dummdidumm** [comment]
Should we put this into `sites` instead? It's somewhat of a site, after all. What are other's opinions on this?

**Comment by ghostdevv** [comment]
> Should we put this into sites instead? It's somewhat of a site, after all. What are other's opinions on this?

The main reason I put it in packages to align with svelte and other monorepo setups in the ecosystem

**Comment by benmccann** [comment]
`packages` does feel a bit off to me. I've seen a lot of repos just put it in the root directory:
- https://github.com/vitejs/vite
- https://github.com/sveltejs/svelte-hmr

**Comment by ghostdevv** [comment]
I like the idea of putting it in the root directory

**Comment by benmccann** [comment]
I think I like @dummdidumm's suggestion of putting it in `sites` actually. It might be a bit more organized that way. And then it'll feel less silly that we have a `sites` directory with only one thing in it :laughing:

**Comment by ghostdevv** [comment]
> I think I like @dummdidumm's suggestion of putting it in sites actually. It might be a bit more organized that way. And then it'll feel less silly that we have a sites directory with only one thing in it laughing

My only worry is that it'll be tucked away - and while it *literally is a site*, it's not really a site in the way I imagine that folder was intended. Putting it in packages has always felt weird to me but I have done it as that's what others done, but if vite is putting it at the root that would be my new preferred solution

**Comment by benmccann** [comment]
In the Svelte 5 codebase we have a few playgrounds that we just moved under a `playgrounds` directory. We should probably do the same here for consistency. This could be called `playgrounds/basic` to match what we've done there where we also have one called `basic`

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,7 +16,15 @@ cd kit
 pnpm install
 ```
 
-You can now run SvelteKit by linking it into your project with [pnpm `overrides`](https://pnpm.io/package_json#pnpmoverrides):
+## Testing Changes
+
+### Playground
+
+You can use the playground at [`playgrounds/basic`](./playgrounds/basic/) to experiment with your changes to Kit locally.
```
Comment (benmccann) [inline CONTRIBUTING.md:23]:
```suggestion
You can use the playground at [`playgrounds/basic`](./playgrounds/basic/) to experiment with your changes to SvelteKit locally.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,7 +16,15 @@ cd kit
...
-You can now run SvelteKit by linking it into your project with [pnpm `overrides`](https://pnpm.io/package_json#pnpmoverrides):
+## Testing Changes
+
+### Playground
+
+You can use the playground at [`playgrounds/basic`](./playgrounds/basic/) to experiment with your changes to SvelteKit locally.
+
+### Linking
+
+If you don't want to use the playground you can use [pnpm `overrides`](https://pnpm.io/package_json#pnpmoverrides) to link it to your existing project:
```
Comment (benmccann) [inline CONTRIBUTING.md:27]:
```suggestion
If you want to test against an existing project, you can use [pnpm `overrides`](https://pnpm.io/package_json#pnpmoverrides) in that project:
```

**Review by benmccann — APPROVED** [review]

**Pull request merged**